tests/server/signupRoute.test.js

```diff
@@ -0,0 +1,71 @@
+import { db } from "../../server/db";
+import Supertest from "supertest";
+
+const App = require("../../server/index");
+const request = Supertest(App);
+
+jest.mock("../../server/db");
+
+const noUNOrPass = "Error: Username or password not set!";
+
+describe("Test signup route", () => {
+  test("Error 400 - no username or password", async () => {
+    const res = await request.post("/api/signup").send({});
+    expect(res.statusCode).toEqual(400);
+    expect(res.text).toEqual(noUNOrPass);
+  });
+
+  test("Error 400 - no password", async () => {
+    const res = await request.post("/api/signup").send({
+      username: "username",
+    });
+    expect(res.statusCode).toEqual(400);
+    expect(res.text).toEqual(noUNOrPass);
+  });
+
+  test("Error 400 - no username", async () => {
+    const res = await request.post("/api/signup").send({
+      password: "password",
+    });
+    expect(res.statusCode).toEqual(400);
+    expect(res.text).toEqual(noUNOrPass);
+  });
+
+  test("Error 401 - bad username", async () => {
+    const res = await request.post("/api/signup").send({
+      username: "un$",
+      password: "Very1Epic!",
+    });
+    expect(res.statusCode).toEqual(401);
+  });
+
+  test("Error 401 - bad password", async () => {
+    const res = await request.post("/api/signup").send({
+      username: "usernameVeryEpic",
+      password: "password",
+    });
+    expect(res.statusCode).toEqual(401);
+  });
+
+  test("Response 200 - happy insertion", (done) => {
+    // this test *may* look a little gross, but this makes sure it hits all the lines of the test
+    request
+      .post("/api/signup")
+      .send({
+        username: "usernameVeryCool",
+        password: "Very1Epic",
+      })
+      .then((res) => {
+        expect(res.statusCode).toEqual(200);
+
+        db.get(
+          `select user_id from User where username="usernameVeryCool"`,
+          function (err, row) {
+            if (err) throw err;
+            expect(row["user_id"]).toBeDefined(); // we don't know the user_id; however, it should be something
+            done();
+          }
+        );
+      });
```

Comment:
It could look a little cleaner by following the same `const res = request(...)` that you have in the other tests. Just pull the db call outside of the `.then()`

Reply:
`done()` is required for callbacks to be handled correctly. You also can't use `done()` within an `async` context; you must either use `done()` in a non-async context or return a Promise (and I'm not sure how to handle that within a callback). So, it'll stay ugly.